reflectometry/detailed_plots.py

```diff
@@ -0,0 +1,108 @@
+if __name__ == '__main__':
+    from collections import OrderedDict
+    from matplotlib import rcParams, cycler
+
+    FONTSIZE = 20
+    NEARLY_BLACK = "#161616"
+    LIGHT_GREY = "#F5F5F5"
+    WHITE = "#ffffff"
+
+    MASTER_FORMATTING = {
+        "axes.formatter.limits": (-3, 3),
+        "xtick.major.pad": 7,
+        "ytick.major.pad": 7,
+        "ytick.color": NEARLY_BLACK,
+        "xtick.color": NEARLY_BLACK,
+        "axes.labelcolor": NEARLY_BLACK,
+        "axes.spines.bottom": True,
+        "axes.spines.left": True,
+        "axes.spines.right": True,
+        "axes.spines.top": True,
+        "axes.axisbelow": True,
+        "legend.frameon": False,
+        'axes.edgecolor': NEARLY_BLACK,
+        "pdf.fonttype": 42,
+        "ps.fonttype": 42,
+        "mathtext.fontset": "custom",
+        "font.size": FONTSIZE,
+        "font.family": "sans-serif",
+        "font.serif" : "Helvetica",
+        #"text.usetex": True,
+        "savefig.bbox": "tight",
+        "axes.facecolor": LIGHT_GREY,
+        "axes.labelpad": 10.0,
+        "axes.labelsize": FONTSIZE * 0.8,
+        "axes.titlepad": 20,
+        "axes.titlesize": FONTSIZE,
+        "axes.grid": False,
+        "grid.color": WHITE,
+        "lines.markersize": 7.0,
+        "lines.scale_dashes": False,
+        "xtick.labelsize": FONTSIZE * 0.8,
+        "ytick.labelsize": FONTSIZE * 0.8,
+        "legend.fontsize": FONTSIZE * 0.8,
+        "lines.linewidth": 2,
+    }
+
+    for k, v in MASTER_FORMATTING.items():
+        rcParams[k] = v
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from scipy.constants import g, h, neutron_mass
+    from scipy.special import erf
+    #import _fig_params
+
+    a = -g
+    v = 20
+    z1 = 4.2
+    y0 = 0
+    y1 = 1
+    t = z1 / v
+
+    def trace(z, z0):
+        return y0 + (
+            (z-z0) * ((-a * (z1 - z0) ** 2) / (2 * v ** 2) - y0 + y1)
+        ) * 1 / (z1 - z0) + (a * (z - z0) ** 2) / (2 * v ** 2)
+
+    def dtrace(z, z0):
+        return ((-a * (z1 - z0) ** 2) / (2 * v ** 2) - y0 + y1) * 1 / (z1 - z0) + a * (z - z0) / (v ** 2)
+
+    z = np.linspace(0.00001, z1-0.01, 1000)
+    z_diff = np.linspace(0.4, z1-0.01, 1000)
+    y = np.linspace(0.00001, y1-0.001, 1000)
+
+    fig, ax = plt.subplots(figsize=(8, 4))
+    plt.plot(z, trace(z, 0))
```

Comment:
I really recommend here to use `ax.plot` instead of `plt.plot`

Reply:
Sob, I like my mess of coding paradigms in `mpl` though 🤣. Sorting now